DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/lldp/test_lldp_receive.py

```diff
@@ -0,0 +1,350 @@
+            'chassisVarLen': len(chassis.replace(':', '')) // 2,
+            'chassisId': chassis.replace(':', ''),
+            'portLen': (len(port.encode().hex()) // 2) + 1,
+            'portSubtype': port_subtype,
+            'portVarLen': len(port.encode().hex()) // 2,
+            'portId': port.encode().hex(),
+            'ttlLen': 2,
+            'ttlVal': ttl,
+    }
+    if scenario == 'ttl':
+        chassis = '00:00:00:11:11:11'
+        port = '00:00:00:00:00:07'
+        ttl = 20
+        port_subtype = 3
+        lldp.update({
+            'chassisId': chassis.replace(':', ''),
+            'portLen': (len(port.replace(':', '')) // 2) + 1,
+            'portSubtype': port_subtype,
+            'portVarLen': len(port.replace(':', '')) // 2,
+            'portId': port.replace(':', ''),
+            'ttlVal': ttl,
+        })
+    elif scenario == 'optional':
+        port = '00:02:03:00:00:07'
+        port_subtype = 3
+        port_desc = 'GigabitEthernet8'
+        sys_name = 'S2.cisco.com'
+        sys_desc = 'Cisco IOS Software, C3560 Software (C3560-ADVIPSERVICESK9-M), Version 12.2(44)SE, RELEASE SOFTWARE (fc1)'
+        mgmt = IPv4Address('10.5.225.52')
+        capabilities = {'Bridge': True, 'Router': False}
+
+        lldp_optional_pkt = LLDP_OPTIONAL_HEX.format(
+            chassis=chassis.replace(':', ''),
+            port=port.replace(':', ''),
+            port_subtype=port_subtype,
+            ttl=hex(ttl)[2:].zfill(4),
+            port_d_len=hex(len(port_desc.encode().hex()) // 2)[2:],
+            port_d=port_desc.encode().hex(),
+            sys_n_len=hex(len(sys_name.encode().hex()) // 2)[2:].zfill(2),
+            sys_n=sys_name.encode().hex(),
+            sys_d_len=hex(len(sys_desc.encode().hex()) // 2)[2:],
+            sys_d=sys_desc.encode().hex(),
+            cap='0014',
+            en_cap='0004',
+            m_ip=mgmt.packed.hex(),
+        )
+
+        lldp_stream = {'lldp_optional': {
+            'type': 'custom',
+            'protocol': '88cc',
+            'ip_source': dev_groups[tg_ports[0]][0]['name'],
+            'ip_destination': dev_groups[tg_ports[1]][0]['name'],
+            'srcMac': 'aa:bb:cc:dd:ee:11',
+            'dstMac': '01:80:c2:00:00:0e',
+            'rate': 2,
+            'frameSize': 512,
+            'transmissionControlType': 'fixedPktCount',
+            'frameCount': 6,
+            'customLength': len(lldp_optional_pkt) // 2,
+            'customData': lldp_optional_pkt}
+        }
+    if scenario != 'optional':
+        lldp_stream = get_lldp_stream(dev_groups[tg_ports[0]][0]['name'], dev_groups[tg_ports[1]][0]['name'], lldp, count=count)
+    await tgen_utils_setup_streams(tgen_dev, config_file_name=None, streams=lldp_stream)
+
+    # Get lldp rx stats from port
+    rx_before = await get_lldp_statistic(dev_name, dut_ports[0], stats='rx')
+    # 4.Transmit lldp packet to first DUT port
+    await tgen_utils_start_traffic(tgen_dev)
+    await asyncio.sleep(10)
```

Comment:
Not sure if 10 seconds is enough since Ixia may be slow sometimes. Can you increase it to e.g. 20 seconds, so we will not see occasional fails due to Ixia performance